lib/SILOptimizer/Mandatory/DiagnoseInfiniteRecursion.cpp

```diff
@@ -94,37 +94,39 @@ static bool hasRecursiveCallInPath(SILBasicBlock &Block,
   return false;
 }
 
-static bool hasInfinitelyRecursiveApply(SILFunction &Fn,
-                                        SILFunction *TargetFn) {
+/// Perform a DFS through the target function to find any paths to an exit node
+/// that do not call into the target.
+static bool hasInfinitelyRecursiveApply(SILFunction *TargetFn) {
```

Comment:
N.B. I had the extra parameter here because I intended to extend this pass to compute mutually-recursive applies in the initializer thunks for lazy vars.